service/mattermost/mattermost_test.go

```diff
@@ -0,0 +1,94 @@
+package mattermost
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const url = "https://host.mattermost.com"
+
+func TestService_New(t *testing.T) {
+	t.Parallel()
+	assert := require.New(t)
+	service := New(url)
+	assert.NotNil(service)
+}
+
+func TestService_LoginWithCredentials(t *testing.T) {
+	t.Parallel()
+	assert := require.New(t)
+	service := New(url)
+	// Test responses
+	mockClient := newMockHttpClient(t)
+	mockClient.
+		On("Send", context.TODO(), "fake-loginID", "fake-password").Return(nil)
+	service.loginClient = mockClient
+	// test call
+	err := service.LoginWithCredentials(context.TODO(), "fake-loginID", "fake-password")
+	assert.Nil(err)
+	mockClient.AssertExpectations(t)
+
+	// Test errors
+	// Test responses
+	mockClient = newMockHttpClient(t)
+	mockClient.
+		On("Send", context.TODO(), "fake-loginID", "").Return(errors.New("empty password"))
+	service.loginClient = mockClient
+	// test call
+	err = service.LoginWithCredentials(context.TODO(), "fake-loginID", "")
+	assert.NotNil(err)
+	mockClient.AssertExpectations(t)
+}
+
+func TestService_AddReceivers(t *testing.T) {
+	t.Parallel()
+
+	assert := require.New(t)
+
+	service := New(url)
+	assert.NotNil(service)
+
+	service.AddReceivers("yfgstwuisnshydhd")
+	assert.Equal(1, len(service.channelIDs))
+
+	service.AddReceivers("yfgstwuisnshydhd", "nwudneyfrwqjs")
+	assert.Equal(3, len(service.channelIDs))
```

Comment:
This isn't a blocking issue by any means, but it occurred to me that this service allows for duplicate channels. Just an oddity I think that notify users will have to keep in mind: if a duplicate channel id is added to the receivers, then the same title and message might show up multiple times in the channels feed. Do you have an opinion about this?

Reply:
Yes, It shows up multiple times and I think there will be no use case for users to send the same message multiple times at the same time.
Handled it by storing channel ids in a map instead of an array.